AuditHistoryExtractor/Classes/AuditHistoryManager.cs

```diff
@@ -47,9 +47,10 @@ private string GetValue(Entity attributeAuditHistoryDetail, string fieldKey)
                     if (optionSetManager == null)
                     {
                         optionSetManager = new OptionSetManager(_service);
-                        optionSetManager.SetupOptionSetValues(attributeAuditHistoryDetail.LogicalName, fieldKey);
                     }
 
+                    optionSetManager.SetupOptionSetValues(attributeAuditHistoryDetail.LogicalName, fieldKey);
```

Comment:
This will reinitialize the OptionSetManager for each option set record, which results in calling the CRM Environment over and over again. 
Would propose my solution (#22) , which will most likely also fix your issue? I have to admit, that I did not 100% understand what you meant.